notebook/tests/selenium/test_dashboard_nav.py

```diff
@@ -3,43 +3,42 @@
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support.ui import WebDriverWait
 from selenium.webdriver.support import expected_conditions as EC
-
 pjoin = os.path.join
 
 def get_list_items(browser):
     return [{
         'link': a.get_attribute('href'),
         'label': a.find_element_by_class_name('item_name').text,
+        'element': a,
     } for a in browser.find_elements_by_class_name('item_link')]
 
 
 def wait_for_selector(browser, selector, timeout=10):
     wait = WebDriverWait(browser, timeout)
     return wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, selector)))
 
-
-
-def test_items(browser, visited=None):
-    tree_root_url = browser.current_url
-    if visited is None:
-        visited = set()
-
-    wait_for_selector(browser, '.item_link')
-    items = get_list_items(browser)
-    print(browser.current_url, len(items))
-    for item in items:
-        print(item)
-        url = item['link']
-        if url.startswith(tree_root_url):
-            print("Going to", url)
-            if url in visited:
-                continue
-            visited.add(url)
-            browser.get(url)
-            wait_for_selector(browser, '.item_link')
-            assert browser.current_url == url
-
-            test_items(browser, visited)
-            #browser.back()
-
-    print()
+def test_items(authenticated_browser):
+    tree_root_url = authenticated_browser.current_url
+    visited_dict = {}
+    # Going down the tree to collect links
+    while True:
+        wait_for_selector(authenticated_browser, '.item_link')
+        items = get_list_items(authenticated_browser)
+        visited_dict[authenticated_browser.current_url] = items
+        print(authenticated_browser.current_url, len(items))
+        if len(items)>1:
```

Comment:
Why does it not work with 1 item? Does it catch the link to go up a directory?

I think it would be good to filter out which items are directories, so that this doesn't accidentally load a notebook or another file.

Reply:
For the first, I was using that to filter out `..` but I see now that that actually introduces weird behaviour at the root.

For the second, I gather that I can distinguish that based on the link target being on the `/tree` vs. `/notebook` or `/file` endpoints?

Reply:
Yep, that's right.

Reply:
Oops! i forgot about this simple fix and have been focused on getting the notebook navigation stuff working on the markdown stuff. I'll fix this quickly